### DISCUSSION TITLE
Open AGB FIRM runs slower than original hardware

### DISCUSSION THREAD

**Discussion by Mii-U** [profi200/open_agb_firm#217] [General]
This can be seen in this youtube video ([https://youtu.be/KM0w4USono0?si=8Uv8bCA3YsbSKKH3]() at 8:58) where VC injections are tested.

I have also tested this side by side with my New 3DS running open AGB FIRM and GBA SP with similar results (the 3ds falls behind) which are noticeable after just a minute.

The exact clock speeds mentioned in the video are found at this link
[https://problemkaputt.de/gbatek.htm#dsbackwardscompatiblegbamode]()

Is it possible for open agb firm to set the clock speed exactly to the original GBA's?

**Comment by profi200** [comment]
Unfortunately this can't be fixed. I have no control of the clock speeds. The only way is to replace the clock crystal. All DS consoles with native GBA mode run a tiny bit slower and the 3DS is no exception.

**Reply by Mii-U** [reply]
Understood. After further testing though, I don't think it affects the actual gameplay. Or at least in a way that I could notice.

Now I am curious as to why Nintendo didn't just make the DS clock exactly 2x the GBA clock since it is so close anyways.

**Reply by profi200** [reply]
That i don't know. Maybe they could not get the exact same clock crystals anymore or had other timing constraints elsewhere.
